centreon/www/widgets/live-top10-cpu-usage/index.php

```diff
@@ -134,12 +134,21 @@
         AND h.enabled = 1
     GROUP BY i.host_id
     ORDER BY current_value DESC
-    LIMIT " . $preferences['nb_lin'] . ";";
+    LIMIT :nb_lin;";
 
 $numLine = 1;
 
-$res = $db->query($query);
-while ($row = $res->fetch()) {
+$statement = $db->prepare($query);
+$statement->bindValue(':service_description', '%' . $preferences['service_description'] . '%', PDO::PARAM_STR);
+$statement->bindValue(':metric_name', '%' . $preferences['metric_name'] . '%', PDO::PARAM_STR);
+if ($preferences['host_group']) {
+    $statement->bindValue(':host_group', $preferences['host_group'], PDO::PARAM_INT);
+}
+$statement->bindValue(':nb_lin', $preferences['nb_lin'], PDO::PARAM_INT);
+
+$statement->execute();
+
+while ($row = $statement->fetch()) {
```

Comment:
* You must to use the new methods of CentreonDB to replace the old methods ;)
* you can catch CentreonDbException to add a business log with CentreonLog :)
* bindValue() is useless except for the limit ;)

Reply:
top ! thanks for your changes :)